tools/lib/h5tools_str.c

```diff
@@ -1074,7 +1074,7 @@ h5tools_str_sprint(h5tools_str_t *str, const h5tool_format_t *info, hid_t contai
                         /* if (H5Tequal(type, H5T_STD_REF_DSETREG)) */
                         H5TOOLS_DEBUG("H5T_REFERENCE:H5T_STD_REF_DSETREG");
                         h5tools_str_append(str, H5_TOOLS_DATASET);
-                        h5tools_str_sprint_reference(str, container, vp);
+                        h5tools_str_sprint_reference(str, container, H5R_DATASET_REGION, vp);
                     }
                     else if (nsize == H5R_OBJ_REF_BUF_SIZE) {
```

Comment:
Don't we have a better way of checking this? It seems really fragile to rely on checking the size here.

Reply:
Of course there is always the H5Tequal call. For 1.10 it is okay because there is only 2 choices.